//
// This file is part of the WoW++ project.
// 
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 2 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with this program; if not, write to the Free Software 
// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
//
// World of Warcraft, and all World of Warcraft or Warcraft art, images,
// and lore are copyrighted by Blizzard Entertainment, Inc.
// 

syntax = "proto2";
package wowpp.proto;

message TriggerExpression {
	// Type. Can be: VALUE, VARIABLE or FUNCTION.
	required uint32 type = 1;
	// Depends on type. If VALUE, this is the value. if VARIABLE, this is the variable id. And if FUNCTION, this is t
	required uint32 value = 2;
	// Only used if type is set to FUNCTION. Values depends on used function.
	repeated uint32 data = 3;
}

message TriggerCondition {
	required uint32 type = 1;
	optional uint32 value_1 = 2;
	optional uint32 value_2 = 3;
}

message TriggerAction {
	optional uint32 action = 1;
	optional uint32 target = 2;
	optional string targetname = 3;
	repeated string texts = 4;
	repeated int32 data = 5;
}

message TriggerEvent {
	required uint32 type = 1;
	repeated uint32 data = 2;
}

message TriggerEntry {
	required uint32 id = 1;
	required string name = 2;
	optional string category = 3;
	reserved 4;
	repeated TriggerAction actions = 5;
	repeated TriggerEvent newevents = 6;
}

message Triggers {
	repeated TriggerEntry entry = 1;
}
